controllers/drplacementcontrol_controller.go

```diff
@@ -1213,24 +1223,272 @@ func (r *DRPlacementControlReconciler) updateResourceCondition(
+//     requiring further processing of vrg for a status decision
+func ProcessVRGClusterDataProtected(drpc *rmn.DRPlacementControl,
+	vrg *rmn.VolumeReplicationGroup,
+	clusterName string,
+) bool {
+	updated := true
+
+	condition := meta.FindStatusCondition(vrg.Status.Conditions, VRGConditionTypeClusterDataProtected)
+
+	if condition != nil && condition.Status == metav1.ConditionTrue && condition.ObservedGeneration == vrg.Generation {
+		return !updated
+	}
+
+	if condition == nil || condition.ObservedGeneration != vrg.Generation || condition.Status == metav1.ConditionUnknown {
+		SetDRPCStatusCondition(&drpc.Status.Conditions, rmn.ConditionProtected, drpc.Generation,
+			metav1.ConditionFalse,
+			rmn.ReasonProtectedProgressing,
+			fmt.Sprintf("VolumeReplicationGroup on cluster %s is progressing on protecting workload resources",
+				clusterName))
+
+		return updated
+	}
+
+	// condition.Status == metav1.ConditionFalse
+	SetDRPCStatusCondition(&drpc.Status.Conditions, rmn.ConditionProtected, drpc.Generation,
+		metav1.ConditionFalse,
+		rmn.ReasonProtectedError,
+		fmt.Sprintf("VolumeReplicationGroup on cluster %s has errors protecting workload resources, "+
+			"retrying till %s condition is met",
+			clusterName, VRGConditionTypeClusterDataProtected))
+
+	return updated
+}
+
+// ProcessVRGDataReadyAndProtected is a helper function used by SetDRPCProtectedCondition
+//   - Returns a bool that is true if status was updated, and false otherwise,
+//     requiring further processing of vrg for a status decision
+func ProcessVRGDataReadyAndProtected(drpc *rmn.DRPlacementControl,
+	vrg *rmn.VolumeReplicationGroup,
+	clusterName string,
+) bool {
+	updated := true
+
+	if vrg.Spec.ReplicationState == rmn.Primary {
+		if ProcessVRGDataReadyAsPrimary(drpc, vrg, clusterName) {
+			return updated
+		}
+
+		return ProcessVRGDataProtectedAsPrimary(drpc, vrg, clusterName)
+	}
+
+	if ProcessVRGDataReadyAsSecondary(drpc, vrg, clusterName) {
+		return updated
+	}
+
+	return ProcessVRGDataProtectedAsSecondary(drpc, vrg, clusterName)
+}
+
+// ProcessVRGDataReadyAsPrimary is a helper function used by ProcessDataReadyAndProtected
+//   - Returns a bool that is true if status was updated, and false otherwise,
+//     requiring further processing of vrg for a status decision
+func ProcessVRGDataReadyAsPrimary(drpc *rmn.DRPlacementControl,
+	vrg *rmn.VolumeReplicationGroup,
+	clusterName string,
+) bool {
+	updated := true
+
+	condition := meta.FindStatusCondition(vrg.Status.Conditions, VRGConditionTypeDataReady)
+
+	if condition != nil && condition.Status == metav1.ConditionTrue && condition.ObservedGeneration == vrg.Generation {
+		return !updated
+	}
+
+	if condition == nil || condition.ObservedGeneration != vrg.Generation || condition.Status == metav1.ConditionUnknown {
+		SetDRPCStatusCondition(&drpc.Status.Conditions, rmn.ConditionProtected, drpc.Generation,
+			metav1.ConditionFalse,
+			rmn.ReasonProtectedProgressing,
```

Comment:
Why is the reason `Progressing`? Shouldn't be Unknown.  Or shouldn't we just take what is in the VRG:
Status: condition.Status
Reason: condition.Reason
Message: condition.Message

Reply:
This should be addressed by the revised commits